segment_os.go

```diff
@@ -26,12 +34,64 @@ func (n *osInfo) string() string {
 	case "darwin":
 		return n.props.getString(MacOS, "\uF179")
 	case "linux":
-		return n.props.getString(Linux, "\uF17C")
+		wsl := n.env.getenv("WSL_DISTRO_NAME")
+		p := n.env.getPlatform()
+		if wsl != "" {
+			return fmt.Sprintf("%s%s%s",
+				n.props.getString(WSL, "WSL"),
+				n.props.getString(WSLSeparator, " - "),
+				n.props.getString(Linux, getLinuxIcon(p)))
+		}
+		return n.props.getString(Linux, getLinuxIcon(p))
 	default:
 		return ""
 	}
 }
 
+func getLinuxIcon(p string) string {
```

Comment:
I would remove this. We have idea if the icon set here will be available in the patched font the user want to use. And adding each and everyone of these as a `Property` is a bit of a bridge too far imho.

Reply:
this assumes nerd fonts, which most of the icons seem to be geared for.  we could get even fancier and do something like p10k where there's a global config setting which type of font is used and sets the default icons appropriately.  https://github.com/romkatv/powerlevel10k/blob/master/internal/icons.zsh

Reply:
I added the distro icon since I thought that had been discussed previously on one of the other PRs

Reply:
If we do go down this path, I feel like we should add them to the Properties as well otherwise you can never override.

On top of that, if so, it might be a good idea to default to icons everywhere and propose a nerd font patched font for max enjoyment.

Reply:
For completeness, that last remark shouldn't be resolved here 😁

Reply:
how about a use simple font vs nerd font setting?  if simple is used, then it defaults to a single linux icon/string property and if nerd font it branches into the distro specific icons?

Reply:
because i agree, having a property for every distro is overload.

Reply:
> how about a use simple font vs nerd font setting?  if simple is used, then it defaults to a single linux icon/string property and if nerd font it branches into the distro specific icons?  

We'd have to be sure all nerd fonts render it consistently then. You're going to slap me, but the more I think about it, the more I feel like adding then as a Property is the least annoying thing to do going forward. And still default to the icons.

Reply:
@nwykes did I demotivate you?

Reply:
lol not at all. Work got crazy and I haven't been able to finish this up.  I've converted the distro icons to properties and just need to put a bow on it and resubmit.